src/main/java/org/opensearch/neuralsearch/util/prune/PruneUtils.java

```diff
@@ -0,0 +1,289 @@
+        float thresh,
+        boolean requiresPrunedEntries
+    ) {
+        Map<String, Float> highScores = new HashMap<>();
+        Map<String, Float> lowScores = requiresPrunedEntries ? new HashMap<>() : null;
+
+        for (Map.Entry<String, Float> entry : sparseVector.entrySet()) {
+            if (entry.getValue() >= thresh) {
+                highScores.put(entry.getKey(), entry.getValue());
+            } else if (requiresPrunedEntries) {
+                lowScores.put(entry.getKey(), entry.getValue());
+            }
+        }
+
+        return new Tuple<>(highScores, lowScores);
+    }
+
+    /**
+     * Prunes a sparse vector by keeping only elements whose cumulative sum of values
+     * is within a certain ratio of the total sum.
+     *
+     * @param sparseVector The input sparse vector as a map of string keys to float values
+     * @param alpha The minimum ratio relative to the total sum for elements to be kept
+     * @param requiresPrunedEntries Whether to return pruned entries
+     * @return A tuple containing two maps: the first with elements meeting the alpha mass threshold,
+     *         the second with remaining elements (or null)
+     */
+    private static Tuple<Map<String, Float>, Map<String, Float>> pruneByAlphaMass(
+        Map<String, Float> sparseVector,
+        float alpha,
+        boolean requiresPrunedEntries
+    ) {
+        List<Map.Entry<String, Float>> sortedEntries = new ArrayList<>(sparseVector.entrySet());
+        sortedEntries.sort(Map.Entry.comparingByValue(Comparator.reverseOrder()));
+
+        float sum = (float) sparseVector.values().stream().mapToDouble(Float::doubleValue).sum();
+        float topSum = 0f;
+
+        Map<String, Float> highScores = new HashMap<>();
+        Map<String, Float> lowScores = requiresPrunedEntries ? new HashMap<>() : null;
+
+        for (Map.Entry<String, Float> entry : sortedEntries) {
+            float value = entry.getValue();
+            topSum += value;
+
+            if (topSum <= alpha * sum) {
+                highScores.put(entry.getKey(), value);
+            } else if (requiresPrunedEntries) {
+                lowScores.put(entry.getKey(), value);
+            }
+        }
+
+        return new Tuple<>(highScores, lowScores);
+    }
+
+    /**
+     * Split a sparse vector using the specified prune type and ratio.
+     *
+     * @param pruneType    The type of prune strategy to use
+     * @param pruneRatio   The ratio or threshold for prune
+     * @param sparseVector The input sparse vector as a map of string keys to float values
+     * @return A tuple containing two maps: the first with high-scoring elements,
+     * the second with low-scoring elements
+     */
+    public static Tuple<Map<String, Float>, Map<String, Float>> splitSparseVector(
+        PruneType pruneType,
+        float pruneRatio,
+        Map<String, Float> sparseVector
+    ) {
+        if (Objects.isNull(pruneType) || Objects.isNull(pruneRatio)) {
```

Comment:
[nit] `Objects.isNull(pruneRatio)` is not necessary as pruneRatio never be null.

Reply:
ack